src/main/java/com/github/manolo8/darkbot/modules/utils/AttackAPIImpl.java

```diff
@@ -1,100 +1,158 @@
+                         ConfigAPI config) {
         super(heroItems, hero);
-        this.main = main;
         this.laserHandler = laserHandler;
+        this.highlight = highlight;
+        this.settingsManager = settingsManager;
+        this.mapManager = mapManager;
+
+//        this.shipAbility = config.requireConfig("loot.ship_ability");
+//        this.shipAbilityHealth = config.requireConfig("loot.ship_ability.min");
     }
 
     @Override
     protected boolean isAttackViaSlotBarEnabled() {
-        return true; //todo
+        return settingsManager.attackViaSlotbar;
     }
 
     @Override
     protected SelectableItem.Laser getBestLaserAmmo() {
         return laserHandler.getBest();
     }
 
-    @Feature(name = "Default laser selector", description = "Default bot ammo selection manager")
-    public class DefaultLaserSupplier implements LaserSelector, PrioritizedSupplier<SelectableItem.Laser> {
-
-        private final ConfigSetting<Boolean> rsbEnabled;
-        private final ConfigSetting<Config.Loot.Sab> sabSettings;
+    @Override
+    public String getStatus() {
+//        String supplierName = Optional.ofNullable(laserHandler.getLastUsedSupplier())
+//                .filter(s -> isLocked())
+//                .map(Object::getClass)
+//                .map(c -> c.getAnnotation(Feature.class))
+//                .map(Feature::name)
+//                .orElse("");
+
+        return hasTarget() ? ("Targeting " + target.getClass().getSimpleName()
+                + " [" + (isLocked() ? "L " : "")
+                + (isAttacking() ? "S " : "")
+                + (isCastingAbility() ? "A " : "")
+                + ("B" + Math.round(buggedTimer.getRemainingFuse() / 1000.0)) + "]")
+                : "Waiting";
+    }
 
-        private long usedRsb;
-        private boolean useRsb, useSab;
+    @Override
+    public boolean isBugged() {
+        return buggedTimer.isInactive();
+    }
 
-        public DefaultLaserSupplier(ConfigAPI config) {
-            this.rsbEnabled = config.requireConfig("loot.rsb.enabled");
-            this.sabSettings = config.requireConfig("loot.sab");
+    @Override
+    public void setTarget(@Nullable Lockable target) {
+        if (this.target != target) {
+            buggedTimer.activate();
+            highLightTimer.disarm();
         }
 
-        @Override
-        public @NotNull PrioritizedSupplier<SelectableItem.Laser> getLaserSupplier() {
-            return this;
-        }
+        super.setTarget(target);
+    }
 
-        @Override
-        public SelectableItem.Laser get() {
-            return useRsb ? SelectableItem.Laser.RSB_75
-                    : useSab ? SelectableItem.Laser.SAB_50
-                    : SelectableItem.Laser.LCB_10;
-            //todo change with config fields
-        }
+    // TODO: 11.07.2022 use offensive abilities on its own (venom/cyborg or diminisher)
```

Comment:
So is this a lost feature atm?

Reply:
yes, but it is kinda replaced by features in plugins